delivery/services/organise_service.py

```diff
@@ -0,0 +1,186 @@
+        self.symlink_project_report(project, organised_project)
+        self.runfolder_service.dump_project_checksums(organised_project)
+        self.runfolder_service.dump_project_samplesheet(runfolder, organised_project)
+        return organised_project
+
+    def symlink_project_report(self, project, organised_project):
+        """
+        Find and symlink the project report to the organised project directory.
+
+        :param project: a Project instance representing the project before organisation
+        :param organised_project: a Project instance representing the project after organisation
+        """
+        project_report_base, project_report_files = self.runfolder_service.get_project_report_files(project)
+
+        def _link_name(target_file):
+            return os.path.join(
+                organised_project.path,
+                self.file_system_service.relpath(
+                    target_file,
+                    project_report_base))
+
+        def _link_path(target_file, link_name):
+            return self.file_system_service.relpath(
+                target_file,
+                self.file_system_service.dirname(link_name)
+            )
+
+        for project_report_file in project_report_files:
+            link_name = _link_name(project_report_file)
+            link_path = _link_path(project_report_file, link_name)
+            self.file_system_service.symlink(link_path, link_name)
+
+    def organise_sample(self, sample, organised_project_path, lanes):
+        """
+        Organise a sample into its own directory under the corresponding project directory. Samples can be excluded
+        from organisation based on which lane they were run on. The sample directory will be named identically to the
+        sample id field. This may be different from the sample name field which is used as a prefix in the file name
+        for the sample files. This is the same behavior as e.g. bcl2fastq uses for sample id and sample name.
+
+        :param sample: a Sample instance representing the sample to be organised
+        :param organised_project_path: the path to the organised project directory under which to place the sample
+        :param lanes: if not None, only samples run on the any of the specified lanes will be organised
+        :return: a new Sample instance representing the sample after organisation
+        """
+
+        def _include_sample_file(f):
+            return not lanes or f.lane_no in lanes
+
+        # symlink each sample in its own directory
+        organised_sample_path = os.path.join(organised_project_path, sample.sample_id)
+
+        # filter the files if lanes should be excluded
+        sample_files_to_symlink = list(filter(_include_sample_file, sample.sample_files))
+        if sample_files_to_symlink:
+            self.file_system_service.makedirs(organised_sample_path)
+
+        # symlink the sample files using relative paths
+        organised_sample_files = []
```

Comment:
I think perhaps this part could benefit from being placed into it's own function called something like `symlink_sample_files` or similar. Now I had to do a double take here to figure out the relationship between `SampleFile` and `Sample` here.

Reply:
:+1: will do